### ISSUE TITLE
AWS SDK V3 support

### ISSUE THREAD

**Issue by cjnoname** [snowflakedb/snowflake-connector-nodejs#365]
Hi,

I want to understand why the AWS SDK v3 changes were reverted two months ago? Now AWS Lambda support NodeJS 18 which supports SDK v3 natively.

Thanks,
CJ

**Comment by netroy** [comment]
@SimbaGithub @sfc-gh-kdama We'd like to also upgrade to a version of snowplow sdk that uses the AWS SDK v3. 
Can you please add more context on why the change was reverted, and if there is anything we can help to move this forward 🙏🏽 .

**Comment by 1davidmichael** [comment]
This warning is now output when using snowflake-connector-nodejs:

```NOTE: The AWS SDK for JavaScript (v2) will be put into maintenance mode in 2023.```

Hopefully this can be addressed before v2 of the AWS SDK is put in maintenance mode.

**Comment by pjprijs** [comment]
same here, please upgrade package to use AWS SDK v3

**Comment by sfc-gh-dszmolka** [comment]
hi, for the warning message
```
(node:39752) NOTE: We are formalizing our plans to enter AWS SDK for JavaScript (v2) into maintenance mode in 2023.

Please migrate your code to use AWS SDK for JavaScript (v3).
For more information, check the migration guide at https://a.co/7PzMCcy
(Use `node --trace-warnings ...` to show where the warning was created)
```

if you wish, you can suppress it with adding
```
require('aws-sdk/lib/maintenance_mode_message').suppress = true;
```
(from comments in https://github.com/aws/aws-sdk-js/blob/master/lib/maintenance_mode_message.js#L14) until we can deal properly with the issue. thank you for bearing with us !

**Comment by robfig** [comment]
This is our last dependency on AWS v2 before we can drop that 85MB dependency from our builds. Excited for it, thank you!

**Comment by rrajaravi** [comment]
same here - this is our last dependency on AWS V2, please upgrade package to use AWS SDK v3

**Comment by joseph-skupniewicz-sp** [comment]
Yep- seeing this warning also. Please upgrade to use AWS SDK v3 thank you!

**Comment by sahilkul95** [comment]
same here - this is our last dependency on AWS v2. Requesting for upgrade to v3 soon please. Or at least please declare the date of release.
Would be a great help. Thanks a lot!

**Comment by sfc-gh-dszmolka** [comment]
although there's still work to do, but this issue is still on the table and we're working on it (#616 ) . thank you everyone for bearing with us while this is implemented - will keep this updated.

**Comment by sahilkul95** [comment]
got now some info on the release: This is probably being planned for the next month release as per https://github.com/snowflakedb/snowflake-connector-nodejs/pull/616#issuecomment-1689302540

**Comment by sfc-gh-dszmolka** [comment]
#616 merged and will be part of the next release, expected towards end of September

**Comment by sfc-gh-dszmolka** [comment]
fix is now out with release [1.9.0](https://github.com/snowflakedb/snowflake-connector-nodejs/releases/tag/v1.9.0) , [available on npm](https://www.npmjs.com/package/snowflake-sdk/v/1.9.0) as well
